### PR TITLE
Set up Python package, example optimization, and SSP first-order tests

### PR DESCRIPTION
**Pull request by smartalecH**
**TL;DR** Here we set up the repo with the first useful set of routines: the original, first-order SSP algorithm.

We mostly port the existing meep code, but use Jax instead of autograd (jax is still maintained...)

We set things up to eventually support a julia package and a python package. We can use e.g. pycall/juliacall, or maintain two separate paths (might be nice for autograd support). We add a core set of routines, a python example, and a test.

## Examples

The example does a simple experiment on a filtered field at $\beta=\infty$:
```
python3 examples/python/simple_smoothing_example.py
```
projection

It then shows that the gradient is nonzero (hard to see from this plot, but the contours are indeed there):
gradient

And does a mini optimization:
optimization

## Tests

The test effectively does all three of these things:
```
python3 -m unittest discover -s tests/python -p 'test_*.py' -v
test_ccsa_optimization_reduces_fom_to_tolerance (test_ssp_first_order.TestSSPFirstOrder.test_ccsa_optimization_reduces_fom_to_tolerance)
Run a mini CCSA optimization and verify FOM reaches target tolerance. ... ok
test_nonzero_gradient_at_infinite_beta (test_ssp_first_order.TestSSPFirstOrder.test_nonzero_gradient_at_infinite_beta)
For beta=inf, verify gradient is nonzero as in the example. ... ok
test_smoke_pipeline (test_ssp_first_order.TestSSPFirstOrder.test_smoke_pipeline)
Smoke test: import package and run tiny filter + projection pipeline. ... ok

----------------------------------------------------------------------
Ran 3 tests in 2.335s

OK
```

### CODE REVIEW

File: examples/python/simple_smoothing_example.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+"""
+This is a simple example that demonstrates how to use the naive first-order accurate
```
Comment (stevengj) [inline examples/python/simple_smoothing_example.py]: I want to be careful to distinguish what is "first-order" here.

There is SSP1 vs. SSP2, but that isn't about "accuracy" so much as whether the definition of SSP is regularized with second-derivative information.

There's also the separate question of whether one is using bilinear interpolation or, e.g., bicubic.

Comment (stevengj) [reply examples/python/simple_smoothing_example.py]: if it's SSP1 with bilinear interpolation, maybe we should just call it `ssp1_bilinear`.

File: examples/python/simple_smoothing_example.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+SSP algorithm. We set up a dummy, gradient-based optimization problem that attempts
```
Comment (stevengj) [inline examples/python/simple_smoothing_example.py]:
```suggestion
This is a simple example that demonstrates how to use the original "SSP1" algorithm
for subpixel-smoothed projection, combined with bilinear interpolation and conic smoothing.
We set up a simplistic gradient-based optimization problem that attempts
```

File: examples/python/simple_smoothing_example.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    """A simple, convex reduction mean as the FOM.
+
+    FOM = 1/(n*m)ΣΣ|x|^2
+    """
+    return jnp.mean((jnp.abs(x) ** 2).flatten())
+
+
+def full_system(x: jnp.ndarray, beta, eta_i, resolution) -> float:
+    """Include the projection and FOM"""
+    return figure_of_merit(ssp_first_order(x, beta, eta_i, resolution))
```
Comment (stevengj) [inline examples/python/simple_smoothing_example.py]: Can we rename to `ssp1_bilinear`?

As I commented above, it's not "first-order accurate".

Comment (stevengj) [reply examples/python/simple_smoothing_example.py]: and rename the files etc?

Comment (smartalecH) [reply examples/python/simple_smoothing_example.py]: fixed, thanks

File: examples/python/simple_smoothing_example.py (hunk 1 above)
Comment (smartalecH) [reply examples/python/simple_smoothing_example.py]: good call -- fixed, thanks

**Comment by smartalecH** [comment]
@stevengj can you take another look?

File: src/python/ssp/core.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    optimization by subpixel-smoothed projection,” Optics Express, vol. 33, 
+    pp. 33620-33642, July 2025. Editor's Pick.
+
+    Args:
+        rho_filtered: The (2D) input design parameters (already filered e.g.
+            with a conic filter).
+        beta: The thresholding parameter in the range [0, inf]. Determines the
+            degree of binarization of the output.
+        eta: The threshold point in the range [0, 1].
+        resolution: resolution of the design grid (not the Meep grid
```
Comment (stevengj) [inline src/python/ssp/core.py]: Shouldn't refer to Meep.

Would be nice to have an `smoothing_radius` parameter that defaults to `0.55/resolution`

Would be nice to have an `erosion_dilation` parameter similar to https://github.com/NanoComp/meep/pull/3087

Comment (smartalecH) [reply src/python/ssp/core.py]: > Shouldn't refer to Meep.

Fixed, thanks.

> Would be nice to have an smoothing_radius parameter that defaults to 0.55/resolution

Fixed thanks.

> Would be nice to have an erosion_dilation parameter similar to https://github.com/NanoComp/meep/pull/3087

Let's add this in a future PR. For one, there are still some sharp edges with the referenced implementation I'd like to smooth out -- and I'd prefer it if that didn't block this
